src/fns.rs

```diff
@@ -741,6 +741,117 @@ pub(crate) fn levenshtein_distance<T: PartialEq, I: Iterator<Item = T>>(s1: I, s
     v0[t.len()]
 }
 
+#[defun]
+pub(crate) fn string_bytes(string: &str) -> i64 {
+    string.len() as i64
+}
+
+#[defun]
+pub(crate) fn string_lessp(string1: &str, string2: &str) -> bool {
+    for (c1, c2) in string1.chars().zip(string2.chars()) {
+        match std::cmp::Ord::cmp(&c1, &c2) {
+            std::cmp::Ordering::Less => return true,
+            std::cmp::Ordering::Greater => return false,
+            _ => {}
+        }
```

Comment:
We can simplify this to:

```rust
        if c1 != c2 {
            return c1 < c2;
        }
```

Or we could rewrite the whole thing in iterator style:

```rust
    match string1.chars().zip(string2.chars()).find(|(c1, c2)| c1 != c2) {
        Some((c1, c2)) => c1 < c2,
        None => string1.len() < string2.len(),
    }
```